curl/effector/src/main.rs

```diff
@@ -0,0 +1,210 @@
+#![allow(improper_ctypes)]
+#![allow(non_snake_case)]
+
+use eyre::Result;
+use marine_rs_sdk::marine;
+use marine_rs_sdk::module_manifest;
+use marine_rs_sdk::MountedBinaryResult;
+use marine_rs_sdk::WasmLoggerBuilder;
+use curl_effector_types::*;
+
+use itertools::Itertools;
+
+module_manifest!();
+
+const CONNECT_TIMEOUT: usize = 5;
+
+pub fn main() {
+    WasmLoggerBuilder::new()
+        .with_log_level(log::LevelFilter::Info)
+        .build()
+        .unwrap();
+}
+
+fn run_curl(mut cmd: Vec<String>) -> Result<String> {
+    let mut default_arguments = vec![format!("--connect-timeout"),
+    format!("{}", CONNECT_TIMEOUT),
+    String::from("--no-progress-meter"),
+    String::from("--retry"),
+    String::from("0")];
+    cmd.append(&mut default_arguments);
+
+    log::debug!("curl arguments: {:?}", cmd);
+    let result = curl(cmd.clone());
+    log::debug!("curl result: {:?}", result.stringify());
+
+    result
+        .into_std()
+        .ok_or(eyre::eyre!(
+            "stdout or stderr contains non valid UTF8 string"
+        ))?
+        .map_err(|e| eyre::eyre!("curl cli call failed \n{:?}: {}", cmd.iter().join(" "), e))
+}
+
+fn format_header_args(headers: &[HttpHeader]) -> Vec<String> {
+    let mut result = Vec::new();
+    for header in headers {
+        result.push("-H".to_string());
+        result.push(format!("{}: {}", header.name, header.value))
+    }
+    result
+}
+
+//
+// curl <url> -X POST
+//      --data @<data_vault_path>
+//      -H <headers[0]> -H <headers[1]> -H ...
+//      -o <output_vault_path>
+//      --connect-timeout CONNECT_TIMEOUT
+//      --no-progress-meter
+//      --retry 0
+#[marine]
+pub fn curl_post(request: CurlRequest, data_vault_path: String) -> CurlResult {
```

Comment:
Why is it called `data_vault_path`? Seems like just `data` to me 🤔

Reply:
Because it's a path to a file with data for the post request, not the data itself

Reply:
Omg. What if we pass any path without checking, will it work? Feels like a problem